### PR TITLE
Create docker release workflows

### PR DESCRIPTION
**Pull request by warrenvw**
Recently, docker related actions were removed from the release workflows since they were not completing as expected. With this PR, a simpler implementation is introduced to build images from the `st2-docker`, `st2-dockerfiles` and `st2enterprise-dockerfiles` repositories.

Remaining work:
 - [x] Support preparing the `master` branch in the `*-dockerfiles` repositories to point at the next version of st2 (usually `x.ydev`).
 - [ ] Update the st2 release process wiki (as soon as this PR is merged).

I will take responsibility for running these workflows as part of the next release, and address any issues that arise.

### CODE REVIEW

File: actions/docker_finalize_release.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+name: docker_finalize_release
+runner_type: orquesta
+description: Finalize docker images for st2 release
```
Comment (blag) [inline actions/docker_finalize_release.yaml]: I would include a few steps that the workflow does, `Finalize (change version and tag) docker images for st2 release`, that way uninitiated people like me can immediately tell what it does. Doesn't have to give an extensive list of all steps or anything, just a quick overview that's a bit more descriptive than the workflow name.

File: actions/docker_finalize_release.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+entry_point: workflows/docker_finalize_release.yaml
+parameters:
+  version:
+    type: string
+    description: Version used for the release. Should include the patch e.g. 0.1.0
+    required: true
+  fork:
+    type: string
+    description: Fork to use
+    default: StackStorm
```
Comment (blag) [inline actions/docker_finalize_release.yaml:15]: As discussed on Slack, the `fork` parameter needs to be renamed and the description should include a blurb about it being a GitHub username or organization name.

File: actions/docker_finalize_release.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  host:
+    type: string
+    description: Host where to run the release process.
+  cwd:
```
Comment (blag) [inline actions/docker_finalize_release.yaml:19]: @m4dcoder Does the Orquesta runner already accept a `cwd` parameter?

File: actions/docker_prep_dev.meta.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+runner_type: orquesta
+description: Prep dev images for st2 release
+enabled: true
+entry_point: workflows/docker_prep_dev.yaml
+parameters:
+  version:
+    type: string
+    description: Version used for the release. Should include the patch e.g. 0.1.0
+    required: true
+  fork:
```
Comment (blag) [inline actions/docker_prep_dev.meta.yaml]: This `fork` parameter should also be tweaked.

**Review by blag — CHANGES_REQUESTED** [review]
Looks good, just some small tweaks to make.

File: actions/docker_finalize_release.yaml (hunk 3 above)
Comment (m4dcoder) [reply actions/docker_finalize_release.yaml:19]: No. If I recall, the cwd param is specific to a number of release actions.

**Review by blag — APPROVED** [review]

**Comment by Kami** [comment]
It looks like finalize docker workflow failed today when finalizing the release:

```bash
parameters: 
  version: 3.0.0
status: failed (1s elapsed)
start_timestamp: Fri, 26 Apr 2019 06:57:34 UTC
end_timestamp: Fri, 26 Apr 2019 06:57:35 UTC
result: 
  errors:
  - message: Action "st2cd.tag_release" has unexpected input "org".
    schema_path: properties.tasks.patternProperties.^\w+$.properties.input.patternProperties.^\w+$
    spec_path: tasks.st2_dockerfiles_tag_release.input.org
    type: content
  - message: Action "st2cd.tag_release" has unexpected input "org".
    schema_path: properties.tasks.patternProperties.^\w+$.properties.input.patternProperties.^\w+$
    spec_path: tasks.st2_docker_tag_release.input.org
    type: content
  - message: Action "st2cd.tag_release" has unexpected input "org".
    schema_path: properties.tasks.patternProperties.^\w+$.properties.input.patternProperties.^\w+$
    spec_path: tasks.st2enterprise_dockerfiles_tag_release.input.org
    type: content
  - expression: v<% ctx().major_minor_version %>
    language: yaql
    message: Variable "major_minor_version" is referenced before assignment.
    schema_path: properties.tasks.patternProperties.^\w+$.properties.input
    spec_path: tasks.prep_st2enterprise_dockerfiles.input
    type: context
```

I'm still dealing with some other things so I will have a look later, if no one gets to it before me.

**Pull request merged**